src/transform/src/demand.rs

```diff
@@ -114,7 +135,36 @@ impl Demand {
 
                     self.action(value, needs, gets)
                 }
-                MirRelationExpr::LetRec { .. } => Err(crate::TransformError::LetRecUnsupported)?,
+                MirRelationExpr::LetRec {
+                    ids,
+                    values,
+                    max_iters: _,
+                    body,
+                } => {
+                    let ids_used_across_iterations = MirRelationExpr::recursive_ids(ids, values)?
+                        .iter()
+                        .map(|id| Id::Local(*id))
+                        .collect::<BTreeSet<_>>();
+                    let ids = ids.iter().map(|id| Id::Local(*id)).collect_vec();
+                    for id in ids.iter() {
+                        let prior = gets.insert(id.clone(), BTreeSet::new());
+                        assert!(prior.is_none()); // no shadowing
+                    }
+                    self.action(body, columns, gets)?;
+                    for (id, value) in ids.iter().rev().zip_eq(values.iter_mut().rev()) {
+                        let needs = if !ids_used_across_iterations.contains(id) {
+                            gets.remove(id).unwrap()
```

Comment:
nit: use `.expect("existing gets entry")` instead of `.unwrap()`.

See https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style.